File: Sources/Gamified/Models/RegisteredAchievement.swift
//
//  RegisteredAchievement.swift
//  
//
//  Created by Nathan Fallet on 03/04/2023.
//

import Foundation

public struct RegisteredAchievement {
    
    /// Key of the achievement
    public var key: String
    
    /// Name of the achievement
    public var name: String
    
    /// Icon shown in the achievement
    public var icon: String
    
    /// Target value to unlock this achievement
    public var target: Int
    
    /// Create a registered achievement
    /// - Parameters:
    ///   - key: Key of the achievement
    ///   - name: Name of the achievement
    ///   - icon: Icon shown in the achievement
    ///   - target: Target value to unlock this achievement
    public init(
        key: String,
        name: String,
        icon: String,
        target: Int
    ) {
        self.key = key
        self.name = name
        self.icon = icon
        self.target = target
    }
    
}

File: Gamified/src/main/java/me/nathanfallet/gamified/models/RegisteredAchievement.kt
package me.nathanfallet.gamified.models

data class RegisteredAchievement(
    /**
     * Key of the achievement
     */
    val key: String,

    /**
     * Name of the achievement
     */
    val name: String,

    /**
     * Icon shown in the achievement
     */
    val icon: String,

    /**
     * Target value to unlock this achievement
     */
    val target: Long,

    /**
     * Experience given when achieved
     */
    val experience: Long
)
